src/test/groovy/io/openliberty/tools/gradle/TestMultiModuleLooseEarWithPages.groovy

```diff
@@ -0,0 +1,124 @@
+        FileInputStream input = new FileInputStream(on);
+        String ejbWar = "/ejb-war-1.0-SNAPSHOT.war"
+        String warWebappsFolder = "/multi-module-loose-ear-pages-test/war/src/main/webapp"
+        String ejbJar = "/WEB-INF/lib/ejb-jar-1.0-SNAPSHOT.jar"
+
+
+        // get input XML Document
+        DocumentBuilderFactory inputBuilderFactory = DocumentBuilderFactory.newInstance();
+        inputBuilderFactory.setIgnoringComments(true);
+        inputBuilderFactory.setCoalescing(true);
+        inputBuilderFactory.setIgnoringElementContentWhitespace(true);
+        inputBuilderFactory.setValidating(false);
+        inputBuilderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false)
+        inputBuilderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false)  
+        DocumentBuilder inputBuilder = inputBuilderFactory.newDocumentBuilder();
+        Document inputDoc=inputBuilder.parse(input);
+
+        // parse input XML Document
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        String expression = "/archive/dir";
+        NodeList nodes = (NodeList) xPath.compile(expression).evaluate(inputDoc, XPathConstants.NODESET);
+        Assert.assertEquals("Number of <dir/> element ==>", 0, nodes.getLength());
+
+        expression = "/archive/archive";
+        nodes = (NodeList) xPath.compile(expression).evaluate(inputDoc, XPathConstants.NODESET);
+        Assert.assertEquals("Number of <archive/> element ==>", 2, nodes.getLength());
+        String expectedResult1 = nodes.item(0).getAttributes()
+                .getNamedItem("targetInArchive").getNodeValue();
+        String expectedResult2 = nodes.item(1).getAttributes()
+                .getNamedItem("targetInArchive").getNodeValue();
+        Assert.assertTrue("actual targetInArchive paths [" + expectedResult1 + ","
+                + expectedResult2 + "] is not matching with actual result : " + ejbWar,
+                expectedResult1.equals(ejbWar) || expectedResult2.equals(ejbWar))
+
+        if (OSUtil.isWindows()) {
+            warWebappsFolder = "\\multi-module-loose-ear-pages-test\\war\\src\\main\\webapp"
+        }
+        expression = "/archive/archive/dir";
+        nodes = (NodeList) xPath.compile(expression).evaluate(inputDoc, XPathConstants.NODESET);
+        Assert.assertEquals("Number of <archive/> element ==>", 3, nodes.getLength());
+        String sourceOnDisk1 = nodes.item(0).getAttributes().getNamedItem("sourceOnDisk").getNodeValue();
+        String sourceOnDisk2 = nodes.item(1).getAttributes().getNamedItem("sourceOnDisk").getNodeValue();
+        String sourceOnDisk3 = nodes.item(2).getAttributes().getNamedItem("sourceOnDisk").getNodeValue();
+
+        Assert.assertTrue("actual targetInArchive paths [" + sourceOnDisk1 + ","+ sourceOnDisk2 + ","
```

Comment:
message is misleading....targetInArchive should say sourceOnDisk.

Reply:
corrected